reflexio/server/services/base_generation/_should_run.py

```
@@ -0,0 +1,299 @@
+            self._collect_scoped_interactions_for_precheck(extractor_config)
+        )
+        if not session_data_models:
+            logger.info(
+                "No interactions found for consolidated should_generate check for %s",
+                self._get_service_name(),
+            )
+            return False
+
+        # Cheap pre-filter: reject batches that are structurally unable
+        # to yield signal (slash-commands only, too-short user turns,
+        # extractor-prompt echoes) without burning a 5–7s LLM call. See
```

Ruff (0.15.20)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)